### PR TITLE
Made the client synchronous

### PR DESCRIPTION
**Pull request by lexecon**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [![Build Status](https://travis-ci.org/reportportal/client-javascript.svg?branch=master)](https://travis-ci.org/reportportal/client-javascript)[![Code Coverage](https://codecov.io/gh/reportportal/client-javascript/branch/master/graph/badge.svg)](https://codecov.io/gh/reportportal/client-javascript)[![npm version](https://badge.fury.io/js/reportportal-client.svg)](https://badge.fury.io/js/reportportal-client)
 
 #ReportPortal js client 
-This is ALPHA version. It works fine, but public API could be changed.
+Client to communicate with the Report Portal on node js.
```
Comment (ksuglotova) [inline README.md]: This Client is to communicate

File: README.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
 
 Library used only for implementors of custom listeners for ReportPortal.
```
Comment (ksuglotova) [inline README.md]: Library is used

File: README.md [hunk 3]
```diff
@@ -11,89 +11,227 @@ Library used only for implementors of custom listeners for ReportPortal.
...
 
-## Usage
+## Settings
+When creating a client instance, you need to specify the following parameters:
+
+Parameter | Description
+--------- | -----------
+token     | user's token Report Portal from which you want to send requests. It can be found on the profile page of this user.
+endpoint  | URL of your server. For example, if you visit the page at 'https://server:8080/ui', then endpoint will be equal to 'https://server:8080/api/v1'.
+launch    | Name of lunch at creation.
```
Comment (ksuglotova) [inline README.md]: Name of launch

File: README.md [hunk 4]
```diff
@@ -11,89 +11,227 @@ Library used only for implementors of custom listeners for ReportPortal.
...
+project   | The name of the project in which the lunches will be created.
```
Comment (ksuglotova) [inline README.md]: in which the launches

File: README.md [hunk 5]
```diff
@@ -11,89 +11,227 @@ Library used only for implementors of custom listeners for ReportPortal.
...
+
+## Api
+Each method (except checkConnect) returns an object of a specific format:
```
Comment (ksuglotova) [inline README.md]: in a specific format?

File: README.md [hunk 6]
```diff
@@ -11,89 +11,227 @@ Library used only for implementors of custom listeners for ReportPortal.
...
+```javascript
+{
+    tempId: '4ds43fs', // generated by the client id for further work with the created item
+    promise: Promise // An object indicating the completion of an operation
+}
+```
+The client works synchronously, so to send the following requests it is not necessary to wait for the end of previous ones.
```
Comment (ksuglotova) [inline README.md]: so it is not necessary to wait for the end of the previous requests to send following ones.

File: README.md [hunk 7]
```diff
@@ -11,89 +11,227 @@ Library used only for implementors of custom listeners for ReportPortal.
...
+
+### checkConnect
+ checkConnect - asynchronous method for verifying the correct connection of the client
```
Comment (ksuglotova) [inline README.md]: for verifying the correctness of the client connection?

File: README.md [hunk 8]
```diff
@@ -11,89 +11,227 @@ Library used only for implementors of custom listeners for ReportPortal.
...
+    console.log(`You are using an account: ${responce.full_name}`);
+}, (error) => {
+    console.log('Error connection to server');
+    console.dir(error);
+});
+```
 
-Main classes are:
+### startLaunch
+startLaunch - starts a new launch, return temp id that you want to use for all items within this lunch.
```
Comment (ksuglotova) [inline README.md]: returns temp id that you may use for all of the launches within this launch

File: README.md [hunk 9]
```diff
@@ -11,89 +11,227 @@ Library used only for implementors of custom listeners for ReportPortal.
...
+});
+console.log(launchObj.tempId);
+```
+The method takes one argument:
+* launch data object:
+
+Parameter | Description
+--------- | -----------
+start_time | (optional) start time launch(unix time). Default: rpClient.helpers.now()
+name      | (optional) lunch name. Default: parameter 'launch' specified when creating the client instance
```
Comment (ksuglotova) [inline README.md]: launch

File: README.md [hunk 10]
```diff
@@ -11,89 +11,227 @@ Library used only for implementors of custom listeners for ReportPortal.
...
+mode      | (optional) "DEFAULT" or "DEBUG". Default: "DEFAULT"
+description | (optional) description of the launch (supports markdown syntax)
+tags      | (optional) array of launch tags
```
Comment (ksuglotova) [inline README.md]: would it be useful to add tags separator in tags parameter description?

File: README.md [hunk 11]
```diff
@@ -11,89 +11,227 @@ Library used only for implementors of custom listeners for ReportPortal.
...
+
+If you want to know the real id launch, wait for the method to finish (the real id does not participate in the client's work)
```
Comment (ksuglotova) [inline README.md]: To know the real launch id wait for the method to finish (the real id is not used by the client)

File: README.md [hunk 12]
```diff
@@ -11,89 +11,227 @@ Library used only for implementors of custom listeners for ReportPortal.
...
+    console.log(`Launch real id: ${responce.id}`);
+}, (error) => {
+    console.dir(`Error at the start of lunch: ${error}`);
+})
+```
 
-- RPClient
+### finishLaunch
+finishLaunch - finish of the launch. After calling this method, you can not add items to the launch.
+The request to finish the lunch will be sent only after all items within it have finished.
```
Comment (ksuglotova) [inline README.md]: launch

File: README.md [hunk 13]
```diff
@@ -11,89 +11,227 @@ Library used only for implementors of custom listeners for ReportPortal.
...
+});
+```
+The method takes two arguments:
+* id launch (returned by method 'startLaunch')
+* data object:
+
+Parameter | Description
+--------- | -----------
+end_time  | (optional) end time of lunch. Default: rpClient.helpers.now()
+status    | (optional) status launch, one of "", "PASSED", "FAILED", "STOPPED", "SKIPPED", "RESTED", "CANCELLED". Default: "".
```
Comment (ksuglotova) [inline README.md]: status of launch

File: README.md [hunk 14]
```diff
@@ -11,89 +11,227 @@ Library used only for implementors of custom listeners for ReportPortal.
...
+    launchObj.tempId, {
+        description: 'new launch description',
+        tags: ['new_tag1', 'new_tag2'],
+        mode: 'DEBUG'
+    }
+);
+```
+The method takes two arguments:
+* id launch (returned by method 'startLaunch')
+* data object - can contain the following fields: description, tags, mode. These fields can be looked up in the method "startLaunch".
```
Comment (ksuglotova) [inline README.md]: may contain

File: README.md [hunk 15]
```diff
@@ -11,89 +11,227 @@ Library used only for implementors of custom listeners for ReportPortal.
...
+```
+The method takes two arguments:
+* id item (returned by method 'startTestItem')
+* data object:
 
-```js
-var Service = require("reportportal-client");
+Parameter | Description
+--------- | -----------
+end_time  | (optional) end time of lunch. Default: rpClient.helpers.now()
```
Comment (ksuglotova) [inline README.md]: launch

File: README.md [hunk 16]
```diff
@@ -11,89 +11,227 @@ Library used only for implementors of custom listeners for ReportPortal.
...
+status    | (optional) status launch, one of "", "PASSED", "FAILED", "STOPPED", "SKIPPED", "RESTED", "CANCELLED". Default: "PASSED".
```
Comment (ksuglotova) [inline README.md]: launch status

File: README.md [hunk 17]
```diff
@@ -11,89 +11,227 @@ Library used only for implementors of custom listeners for ReportPortal.
...
+})
 ```
+The method takes two arguments:
+* id item (returned by method 'startTestItem')
+* data object:
+
+Parameter | Description
+--------- | -----------
+time      | (optional) time of log. Default: rpClient.helpers.now()
+message   | (optional) message log. Default: ''.
```
Comment (ksuglotova) [inline README.md]: log message

File: README.md [hunk 18]
```diff
@@ -11,89 +11,227 @@ Library used only for implementors of custom listeners for ReportPortal.
...
+status    | (optional) status log, one of 'trace', 'debug', 'info', 'warn', 'error', ''. Default "".
```
Comment (ksuglotova) [inline README.md]: log status

**Review by ksuglotova — CHANGES_REQUESTED** [review]

**Comment by Xotabu4** [comment]
Thanks for documentation!

File: README.md (hunk 8 above)
Comment (lexecon) [reply README.md]: for all *items*

File: README.md (hunk 10 above)
Comment (lexecon) [reply README.md]: do not understand

**Pull request merged**